lms/djangoapps/instructor_task/models.py

```diff
@@ -300,12 +300,18 @@ def store(self, course_id, filename, buff, batched=False):
         path = self.path_to(course_id, filename)
         # See https://github.com/boto/boto/issues/2868
         # Boto doesn't play nice with unicod in python3
-        if not six.PY2 and not batched:
-            buff = ContentFile(buff.read().encode('utf-8'))
-
-        self.storage.save(path, buff)
-
-    def store_rows(self, course_id, filename, rows, encode_for_utf8=False, batched=False):
+        if batched:
+            output_buffer = ContentFile(b'')
+            # for large file buff.read() throws memory error so we have to use chunks
+            for chunk in buff.chunks():
+                output_buffer.write(chunk.encode('utf-8'))
+            self.storage.save(path, output_buffer)
+        else:
```

Comment:
It seems to be redundant code for the export stats feature i.e [ProblemResponses](https://github.com/edx-solutions/edx-platform/blob/release-candidate/lms/djangoapps/instructor_task/tasks_helper/grades.py#L1020-L1023)

Reply:
[add_rows](https://github.com/edx-solutions/edx-platform/blob/release-candidate/lms/djangoapps/instructor_task/models.py#L275) method almost doing the same, Please consider it against what you are doing.